radix-engine/src/updates/cuttlefish.rs

```diff
@@ -160,3 +174,86 @@ fn generate_cuttlefish_transaction_validation_updates() -> StateUpdates {
         ),
     )
 }
+
+fn generate_cuttlefish_metadata_fix<S: SubstateDatabase + ?Sized>(db: &S) -> StateUpdates {
+    struct MetadataUpdates {
```

Comment:
MINOR: Instead of this, it might be more standard to use `MetadataInit` and the `metadata_init! { }` macro below - which would let us do `description: "Hello"` instead of `description: Some(MetadataValue::String("Hello".into()))`  :)

For each node, we could then use `MetadataNativePackage::init_system_struct(init)` to validate and extract the kv entries to `IndexMap<Vec<u8>, KVEntry>` (or even take out a `MetadataNativePackage::map_metadata_init(init) -> Result<IndexMap<Vec<u8>, KVEntry>, ...>` method from inside `init_system_struct` to make it cleaner).

And then just map `IndexMap<Vec<u8>, KVEntry>` to `set_substate` calls.

Reply:
In my opinion, from a code readability / verification standpoint, checking the keys are right is easier than extracting a name from `Some(MetadataValue::String("Hello".into()))`, and ignoring lots of entries with no descriptions - but fair enough.